// @dinari/api-sdk/resources/v2/accounts/accounts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Accounts = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const ActivitiesAPI = tslib_1.__importStar(require("./activities.js"));
const activities_1 = require("./activities.js");
const OrderFulfillmentsAPI = tslib_1.__importStar(require("./order-fulfillments.js"));
const order_fulfillments_1 = require("./order-fulfillments.js");
const OrdersAPI = tslib_1.__importStar(require("./orders.js"));
const orders_1 = require("./orders.js");
const TokenTransfersAPI = tslib_1.__importStar(require("./token-transfers.js"));
const token_transfers_1 = require("./token-transfers.js");
const WithdrawalRequestsAPI = tslib_1.__importStar(require("./withdrawal-requests.js"));
const withdrawal_requests_1 = require("./withdrawal-requests.js");
const WithdrawalsAPI = tslib_1.__importStar(require("./withdrawals.js"));
const withdrawals_1 = require("./withdrawals.js");
const OrderRequestsAPI = tslib_1.__importStar(require("./order-requests/order-requests.js"));
const order_requests_1 = require("./order-requests/order-requests.js");
const WalletAPI = tslib_1.__importStar(require("./wallet/wallet.js"));
const wallet_1 = require("./wallet/wallet.js");
const headers_1 = require("../../../internal/headers.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * **`Accounts` represent the financial accounts of an `Entity`.**
 *
 * `Orders`, dividends, and other transactions are associated with an `Account`.
 */
class Accounts extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.wallet = new WalletAPI.WalletResource(this._client);
        this.orders = new OrdersAPI.Orders(this._client);
        this.orderFulfillments = new OrderFulfillmentsAPI.OrderFulfillments(this._client);
        this.orderRequests = new OrderRequestsAPI.OrderRequests(this._client);
        this.withdrawalRequests = new WithdrawalRequestsAPI.WithdrawalRequests(this._client);
        this.withdrawals = new WithdrawalsAPI.Withdrawals(this._client);
        this.tokenTransfers = new TokenTransfersAPI.TokenTransfers(this._client);
        this.activities = new ActivitiesAPI.Activities(this._client);
    }
    /**
     * Get a specific `Account` by its ID.
     *
     * @example
     * ```ts
     * const account = await client.v2.accounts.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(accountID, options) {
        return this._client.get((0, path_1.path) `/api/v2/accounts/${accountID}`, options);
    }
    /**
     * Set the `Account` to be inactive. Inactive accounts cannot be used for trading.
     *
     * @example
     * ```ts
     * const account = await client.v2.accounts.deactivate(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    deactivate(accountID, options) {
        return this._client.post((0, path_1.path) `/api/v2/accounts/${accountID}/deactivate`, options);
    }
    /**
     * Get the cash balances of the `Account`, including stablecoins and other cash
     * equivalents.
     *
     * @example
     * ```ts
     * const response = await client.v2.accounts.getCashBalances(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    getCashBalances(accountID, options) {
        return this._client.get((0, path_1.path) `/api/v2/accounts/${accountID}/cash`, options);
    }
    /**
     * Get dividend payments made to the `Account` from dividend-bearing stock
     * holdings.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.getDividendPayments(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { end_date: '2019-12-27', start_date: '2019-12-27' },
     *   );
     * ```
     */
    getDividendPayments(accountID, query, options) {
        return this._client.get((0, path_1.path) `/api/v2/accounts/${accountID}/dividend_payments`, { query, ...options });
    }
    /**
     * Get interest payments made to the `Account` from yield-bearing cash holdings.
     *
     * Currently, the only yield-bearing stablecoin accepted by Dinari is
     * [USD+](https://usd.dinari.com/).
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.getInterestPayments(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { end_date: '2019-12-27', start_date: '2019-12-27' },
     *   );
     * ```
     */
    getInterestPayments(accountID, query, options) {
        return this._client.get((0, path_1.path) `/api/v2/accounts/${accountID}/interest_payments`, { query, ...options });
    }
    /**
     * Get the portfolio of the `Account`, excluding cash equivalents such as
     * stablecoins.
     *
     * @example
     * ```ts
     * const response = await client.v2.accounts.getPortfolio(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    getPortfolio(accountID, query = {}, options) {
        return this._client.get((0, path_1.path) `/api/v2/accounts/${accountID}/portfolio`, { query, ...options });
    }
    /**
     * Mints 1,000 mockUSD sandbox payment tokens to the `Wallet` connected to the
     * `Account`.
     *
     * This feature is only supported in sandbox mode.
     *
     * @example
     * ```ts
     * await client.v2.accounts.mintSandboxTokens(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    mintSandboxTokens(accountID, body, options) {
        return this._client.post((0, path_1.path) `/api/v2/accounts/${accountID}/faucet`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Accounts = Accounts;
Accounts.WalletResource = wallet_1.WalletResource;
Accounts.Orders = orders_1.Orders;
Accounts.OrderFulfillments = order_fulfillments_1.OrderFulfillments;
Accounts.OrderRequests = order_requests_1.OrderRequests;
Accounts.WithdrawalRequests = withdrawal_requests_1.WithdrawalRequests;
Accounts.Withdrawals = withdrawals_1.Withdrawals;
Accounts.TokenTransfers = token_transfers_1.TokenTransfers;
Accounts.Activities = activities_1.Activities;
