// @dinari/api-sdk/resources/v2/accounts/accounts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as ActivitiesAPI from "./activities.mjs";
import { Activities } from "./activities.mjs";
import * as OrderFulfillmentsAPI from "./order-fulfillments.mjs";
import { OrderFulfillments, } from "./order-fulfillments.mjs";
import * as OrdersAPI from "./orders.mjs";
import { Orders, } from "./orders.mjs";
import * as TokenTransfersAPI from "./token-transfers.mjs";
import { TokenTransfers, } from "./token-transfers.mjs";
import * as WithdrawalRequestsAPI from "./withdrawal-requests.mjs";
import { WithdrawalRequests, } from "./withdrawal-requests.mjs";
import * as WithdrawalsAPI from "./withdrawals.mjs";
import { Withdrawals, } from "./withdrawals.mjs";
import * as OrderRequestsAPI from "./order-requests/order-requests.mjs";
import { OrderRequests, } from "./order-requests/order-requests.mjs";
import * as WalletAPI from "./wallet/wallet.mjs";
import { WalletResource } from "./wallet/wallet.mjs";
import { buildHeaders } from "../../../internal/headers.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * **`Accounts` represent the financial accounts of an `Entity`.**
 *
 * `Orders`, dividends, and other transactions are associated with an `Account`.
 */
export class Accounts extends APIResource {
    constructor() {
        super(...arguments);
        this.wallet = new WalletAPI.WalletResource(this._client);
        this.orders = new OrdersAPI.Orders(this._client);
        this.orderFulfillments = new OrderFulfillmentsAPI.OrderFulfillments(this._client);
        this.orderRequests = new OrderRequestsAPI.OrderRequests(this._client);
        this.withdrawalRequests = new WithdrawalRequestsAPI.WithdrawalRequests(this._client);
        this.withdrawals = new WithdrawalsAPI.Withdrawals(this._client);
        this.tokenTransfers = new TokenTransfersAPI.TokenTransfers(this._client);
        this.activities = new ActivitiesAPI.Activities(this._client);
    }
    /**
     * Get a specific `Account` by its ID.
     *
     * @example
     * ```ts
     * const account = await client.v2.accounts.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieve(accountID, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}`, options);
    }
    /**
     * Set the `Account` to be inactive. Inactive accounts cannot be used for trading.
     *
     * @example
     * ```ts
     * const account = await client.v2.accounts.deactivate(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    deactivate(accountID, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/deactivate`, options);
    }
    /**
     * Get the cash balances of the `Account`, including stablecoins and other cash
     * equivalents.
     *
     * @example
     * ```ts
     * const response = await client.v2.accounts.getCashBalances(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    getCashBalances(accountID, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/cash`, options);
    }
    /**
     * Get dividend payments made to the `Account` from dividend-bearing stock
     * holdings.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.getDividendPayments(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { end_date: '2019-12-27', start_date: '2019-12-27' },
     *   );
     * ```
     */
    getDividendPayments(accountID, query, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/dividend_payments`, { query, ...options });
    }
    /**
     * Get interest payments made to the `Account` from yield-bearing cash holdings.
     *
     * Currently, the only yield-bearing stablecoin accepted by Dinari is
     * [USD+](https://usd.dinari.com/).
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.getInterestPayments(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { end_date: '2019-12-27', start_date: '2019-12-27' },
     *   );
     * ```
     */
    getInterestPayments(accountID, query, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/interest_payments`, { query, ...options });
    }
    /**
     * Get the portfolio of the `Account`, excluding cash equivalents such as
     * stablecoins.
     *
     * @example
     * ```ts
     * const response = await client.v2.accounts.getPortfolio(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    getPortfolio(accountID, query = {}, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/portfolio`, { query, ...options });
    }
    /**
     * Mints 1,000 mockUSD sandbox payment tokens to the `Wallet` connected to the
     * `Account`.
     *
     * This feature is only supported in sandbox mode.
     *
     * @example
     * ```ts
     * await client.v2.accounts.mintSandboxTokens(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    mintSandboxTokens(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/faucet`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Accounts.WalletResource = WalletResource;
Accounts.Orders = Orders;
Accounts.OrderFulfillments = OrderFulfillments;
Accounts.OrderRequests = OrderRequests;
Accounts.WithdrawalRequests = WithdrawalRequests;
Accounts.Withdrawals = Withdrawals;
Accounts.TokenTransfers = TokenTransfers;
Accounts.Activities = Activities;
