// @dinari/api-sdk/resources/v2/accounts/orders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * **`Orders` represent the buying and selling of assets under an `Account`.**
 *
 * For `Accounts` using self-custodied `Wallets`, `Orders` are created and fulfilled by making calls to Dinari's smart contracts, or using the *Proxied Orders* methods.
 *
 * For `Accounts` using managed `Wallets`, `Orders` are created and fulfilled by using the `Managed Orders` methods, which then create the corresponding transactions on the blockchain.
 */
export class Orders extends APIResource {
    /**
     * Get a specific `Order` by its ID.
     *
     * @example
     * ```ts
     * const order = await client.v2.accounts.orders.retrieve(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { account_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    retrieve(orderID, params, options) {
        const { account_id } = params;
        return this._client.get(path `/api/v2/accounts/${account_id}/orders/${orderID}`, options);
    }
    /**
     * Get a list of all `Orders` under the `Account`. Optionally `Orders` can be
     * filtered by chain ID, transaction hash, or client order ID.
     *
     * @example
     * ```ts
     * const orders = await client.v2.accounts.orders.list(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    list(accountID, query = {}, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/orders`, { query, ...options });
    }
    /**
     * Cancel multiple `Orders` by their IDs in a single request. Note that this
     * requires the `Order` IDs, not the `OrderRequest` IDs. Once you submit a
     * cancellation request, it cannot be undone. Be advised that orders with a status
     * of PENDING_FILL, PENDING_ESCROW, FILLED, REJECTED, or CANCELLED cannot be
     * cancelled.
     *
     * `Order` cancellation is not guaranteed nor is it immediate. The `Orders` may
     * still be executed if the cancellation request is not received in time.
     *
     * The response will indicate which orders were successfully queued to cancel and
     * which failed to queue. Check the status using the "Get Order by ID" endpoint to
     * confirm whether individual `Orders` have been cancelled.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.orders.batchCancel(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { order_ids: ['182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e'] },
     *   );
     * ```
     */
    batchCancel(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/orders/cancel`, { body, ...options });
    }
    /**
     * Cancel an `Order` by its ID. Note that this requires the `Order` ID, not the
     * `OrderRequest` ID. Once you submit a cancellation request, it cannot be undone.
     * Be advised that orders with a status of PENDING_FILL, PENDING_ESCROW, FILLED,
     * REJECTED, or CANCELLED cannot be cancelled.
     *
     * `Order` cancellation is not guaranteed nor is it immediate. The `Order` may
     * still be executed if the cancellation request is not received in time.
     *
     * Check the status using the "Get Order by ID" endpoint to confirm whether the
     * `Order` has been cancelled.
     *
     * @example
     * ```ts
     * const order = await client.v2.accounts.orders.cancel(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   { account_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     * );
     * ```
     */
    cancel(orderID, params, options) {
        const { account_id } = params;
        return this._client.post(path `/api/v2/accounts/${account_id}/orders/${orderID}/cancel`, options);
    }
    /**
     * Get `OrderFulfillments` for a specific `Order`.
     *
     * @example
     * ```ts
     * const fulfillments =
     *   await client.v2.accounts.orders.getFulfillments(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    getFulfillments(orderID, params, options) {
        const { account_id, ...query } = params;
        return this._client.get(path `/api/v2/accounts/${account_id}/orders/${orderID}/fulfillments`, {
            query,
            ...options,
        });
    }
}
