// @dinari/api-sdk/resources/v2/accounts/order-requests/order-requests.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as Eip155API from "./eip155.mjs";
import { Eip155, } from "./eip155.mjs";
import { path } from "../../../../internal/utils/path.mjs";
export class OrderRequests extends APIResource {
    constructor() {
        super(...arguments);
        this.eip155 = new Eip155API.Eip155(this._client);
    }
    /**
     * Get a specific `OrderRequest` by its ID.
     *
     * @example
     * ```ts
     * const orderRequest =
     *   await client.v2.accounts.orderRequests.retrieve(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { account_id: '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e' },
     *   );
     * ```
     */
    retrieve(orderRequestID, params, options) {
        const { account_id } = params;
        return this._client.get(path `/api/v2/accounts/${account_id}/order_requests/${orderRequestID}`, options);
    }
    /**
     * Lists `OrderRequests`. Optionally `OrderRequests` can be filtered by certain
     * parameters.
     *
     * @example
     * ```ts
     * const orderRequests =
     *   await client.v2.accounts.orderRequests.list(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    list(accountID, query = {}, options) {
        return this._client.get(path `/api/v2/accounts/${accountID}/order_requests`, { query, ...options });
    }
    /**
     * Create a managed `OrderRequest` to place a limit buy `Order`.
     *
     * Fees for the `Order` are included in the transaction. Refer to our
     * [Fee Quote API](https://docs.dinari.com/reference/createproxiedorderfeequote#/)
     * for fee estimation.
     *
     * If an `OrderRequest` with the same `client_order_id` already exists for the
     * given account, the creation call will fail.
     *
     * @example
     * ```ts
     * const orderRequest =
     *   await client.v2.accounts.orderRequests.createLimitBuy(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { asset_quantity: 0, limit_price: 0 },
     *   );
     * ```
     */
    createLimitBuy(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/order_requests/limit_buy`, {
            body,
            ...options,
        });
    }
    /**
     * Create a managed `OrderRequest` to place a limit sell `Order`.
     *
     * Fees for the `Order` are included in the transaction. Refer to our
     * [Fee Quote API](https://docs.dinari.com/reference/createproxiedorderfeequote#/)
     * for fee estimation.
     *
     * If an `OrderRequest` with the same `client_order_id` already exists for the
     * given account, the creation call will fail.
     *
     * @example
     * ```ts
     * const orderRequest =
     *   await client.v2.accounts.orderRequests.createLimitSell(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { asset_quantity: 0, limit_price: 0 },
     *   );
     * ```
     */
    createLimitSell(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/order_requests/limit_sell`, {
            body,
            ...options,
        });
    }
    /**
     * Create a managed `OrderRequest` to place a market buy `Order`.
     *
     * Fees for the `Order` are included in the transaction. Refer to our
     * [Fee Quote API](https://docs.dinari.com/reference/createproxiedorderfeequote#/)
     * for fee estimation.
     *
     * If an `OrderRequest` with the same `client_order_id` already exists for the
     * given account, the creation call will fail.
     *
     * @example
     * ```ts
     * const orderRequest =
     *   await client.v2.accounts.orderRequests.createMarketBuy(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { payment_amount: 0 },
     *   );
     * ```
     */
    createMarketBuy(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/order_requests/market_buy`, {
            body,
            ...options,
        });
    }
    /**
     * Create a managed `OrderRequest` to place a market sell `Order`.
     *
     * Fees for the `Order` are included in the transaction. Refer to our
     * [Fee Quote API](https://docs.dinari.com/reference/createproxiedorderfeequote#/)
     * for fee estimation.
     *
     * If an `OrderRequest` with the same `client_order_id` already exists for the
     * given account, the creation call will fail.
     *
     * @example
     * ```ts
     * const orderRequest =
     *   await client.v2.accounts.orderRequests.createMarketSell(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { asset_quantity: 0 },
     *   );
     * ```
     */
    createMarketSell(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/order_requests/market_sell`, {
            body,
            ...options,
        });
    }
    /**
     * Get fee quote data for an `Order Request`. This is provided primarily for
     * informational purposes.
     *
     * For market buy orders, the notional amount of the order includes the fees. For
     * market and limit sell orders, fees are deducted from the proceeds of the sale.
     * For limit buy orders, the fees are added to the total cost of the order.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.accounts.orderRequests.getFeeQuote(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { order_side: 'BUY', order_type: 'MARKET' },
     *   );
     * ```
     */
    getFeeQuote(accountID, body, options) {
        return this._client.post(path `/api/v2/accounts/${accountID}/order_requests/fee_quote`, {
            body,
            ...options,
        });
    }
}
OrderRequests.Eip155 = Eip155;
