// @dinari/api-sdk/resources/v2/market-data/stocks/stocks.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../../core/resource.mjs";
import * as SplitsAPI from "./splits.mjs";
import { Splits, } from "./splits.mjs";
import { path } from "../../../../internal/utils/path.mjs";
export class Stocks extends APIResource {
    constructor() {
        super(...arguments);
        this.splits = new SplitsAPI.Splits(this._client);
    }
    /**
     * Get a list of `Stocks`.
     *
     * @example
     * ```ts
     * const stocks = await client.v2.marketData.stocks.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/api/v2/market_data/stocks/', { query, ...options });
    }
    /**
     * Get current price for a specified `Stock`.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.marketData.stocks.retrieveCurrentPrice(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveCurrentPrice(stockID, options) {
        return this._client.get(path `/api/v2/market_data/stocks/${stockID}/current_price`, options);
    }
    /**
     * Get quote for a specified `Stock`.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.marketData.stocks.retrieveCurrentQuote(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveCurrentQuote(stockID, options) {
        return this._client.get(path `/api/v2/market_data/stocks/${stockID}/current_quote`, options);
    }
    /**
     * Get a list of announced stock dividend details for a specified `Stock`.
     *
     * Note that this data applies only to actual stocks. Yield received for holding
     * tokenized shares may differ from this.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.marketData.stocks.retrieveDividends(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveDividends(stockID, options) {
        return this._client.get(path `/api/v2/market_data/stocks/${stockID}/dividends`, options);
    }
    /**
     * Get historical price data for a specified `Stock`. Each index in the array
     * represents a single tick in a price chart.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.marketData.stocks.retrieveHistoricalPrices(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *     { timespan: 'DAY' },
     *   );
     * ```
     */
    retrieveHistoricalPrices(stockID, query, options) {
        return this._client.get(path `/api/v2/market_data/stocks/${stockID}/historical_prices/`, {
            query,
            ...options,
        });
    }
    /**
     * Get the most recent news articles relating to a `Stock`, including a summary of
     * the article and a link to the original source.
     *
     * @example
     * ```ts
     * const response =
     *   await client.v2.marketData.stocks.retrieveNews(
     *     '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     *   );
     * ```
     */
    retrieveNews(stockID, query = {}, options) {
        return this._client.get(path `/api/v2/market_data/stocks/${stockID}/news`, { query, ...options });
    }
}
Stocks.Splits = Splits;
