// @dinari/api-sdk/resources/v2/entities/entities.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as AccountsAPI from "./accounts.mjs";
import { Accounts, } from "./accounts.mjs";
import * as KYCAPI from "./kyc/kyc.mjs";
import { KYC, } from "./kyc/kyc.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * **`Entities` represent a business or organization that uses the API, and their customers.**
 *
 * Dinari Partners are represented as an organization `Entity` in the API, with their own accounts.
 * Individual customers of Partner `Entities` are also represented as `Entities` in the API, which are managed by the Partner `Entity`.
 */
export class Entities extends APIResource {
    constructor() {
        super(...arguments);
        this.accounts = new AccountsAPI.Accounts(this._client);
        this.kyc = new KYCAPI.KYC(this._client);
    }
    /**
     * Create a new `Entity` to be managed by your organization. This `Entity`
     * represents an individual customer of your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.create({
     *   name: 'x',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/api/v2/entities/', { body, ...options });
    }
    /**
     * Update a specific customer `Entity` of your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    update(entityID, body, options) {
        return this._client.patch(path `/api/v2/entities/${entityID}`, { body, ...options });
    }
    /**
     * Get a list of direct `Entities` your organization manages. These `Entities`
     * represent individual customers of your organization.
     *
     * @example
     * ```ts
     * const entities = await client.v2.entities.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/api/v2/entities/', { query, ...options });
    }
    /**
     * Get a specific customer `Entity` of your organization by their ID.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.retrieveByID(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieveByID(entityID, options) {
        return this._client.get(path `/api/v2/entities/${entityID}`, options);
    }
    /**
     * Get the current authenticated `Entity`, which represents your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.retrieveCurrent();
     * ```
     */
    retrieveCurrent(options) {
        return this._client.get('/api/v2/entities/me', options);
    }
}
Entities.Accounts = Accounts;
Entities.KYC = KYC;
