// @dinari/api-sdk/resources/v2/entities/entities.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Entities = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const AccountsAPI = tslib_1.__importStar(require("./accounts.js"));
const accounts_1 = require("./accounts.js");
const KYCAPI = tslib_1.__importStar(require("./kyc/kyc.js"));
const kyc_1 = require("./kyc/kyc.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * **`Entities` represent a business or organization that uses the API, and their customers.**
 *
 * Dinari Partners are represented as an organization `Entity` in the API, with their own accounts.
 * Individual customers of Partner `Entities` are also represented as `Entities` in the API, which are managed by the Partner `Entity`.
 */
class Entities extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.accounts = new AccountsAPI.Accounts(this._client);
        this.kyc = new KYCAPI.KYC(this._client);
    }
    /**
     * Create a new `Entity` to be managed by your organization. This `Entity`
     * represents an individual customer of your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.create({
     *   name: 'x',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/api/v2/entities/', { body, ...options });
    }
    /**
     * Update a specific customer `Entity` of your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.update(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    update(entityID, body, options) {
        return this._client.patch((0, path_1.path) `/api/v2/entities/${entityID}`, { body, ...options });
    }
    /**
     * Get a list of direct `Entities` your organization manages. These `Entities`
     * represent individual customers of your organization.
     *
     * @example
     * ```ts
     * const entities = await client.v2.entities.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/api/v2/entities/', { query, ...options });
    }
    /**
     * Get a specific customer `Entity` of your organization by their ID.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.retrieveByID(
     *   '182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e',
     * );
     * ```
     */
    retrieveByID(entityID, options) {
        return this._client.get((0, path_1.path) `/api/v2/entities/${entityID}`, options);
    }
    /**
     * Get the current authenticated `Entity`, which represents your organization.
     *
     * @example
     * ```ts
     * const entity = await client.v2.entities.retrieveCurrent();
     * ```
     */
    retrieveCurrent(options) {
        return this._client.get('/api/v2/entities/me', options);
    }
}
exports.Entities = Entities;
Entities.Accounts = accounts_1.Accounts;
Entities.KYC = kyc_1.KYC;
